example/spec/parallel/startSF/startSfSpec.js

```diff
@@ -274,94 +238,25 @@ describe('the sf-starter lambda function', () => {
       ]);
     });
 
-    it('queue-granules returns the correct amount of queued executions', async () => {
-      const granules = new Array(numberOfMessages)
-        .fill()
-        .map((value) => ({
-          granuleId: `granule${value}`,
-          dataType: collection.dataType,
-          version: collection.version,
-          files: []
-        }));
-
-      const response = await lambda().invoke({
-        FunctionName: `${config.stackName}-QueueGranules`,
-        InvocationType: 'RequestResponse',
-        Payload: JSON.stringify({
-          cumulus_meta: {
-            message_source: 'local',
-            execution_name: 'test-execution',
-            task: 'QueueGranules'
-          },
-          meta: {
-            stack: config.stackName,
-            buckets: {
-              internal: {
-                name: config.bucket
-              }
-            },
-            templates: {
-              [waitPassSfName]: templateUri
-            },
-            provider: {},
-            queues: {
-              [maxQueueName]: maxQueueUrl
-            }
-          },
-          workflow_config: {
-            QueueGranules: {
-              stackName: '{{$.meta.stack}}',
-              queueUrl: `{{$.meta.queues.${maxQueueName}}}`,
-              granuleIngestMessageTemplateUri: `{{$.meta.templates.${waitPassSfName}}}`,
-              provider: '{{$.meta.provider}}',
-              internalBucket: '{{$.meta.buckets.internal.name}}'
-            }
-          },
-          payload: {
-            granules: granules
-          }
-        })
-      }).promise();
-      const { Payload } = response;
-
-      console.log(`request ID: ${response.$response.requestId}`);
-
-      const { payload } = JSON.parse(Payload);
-      expect(payload.queued.length).toEqual(numberOfMessages);
-    });
-
     it('consumes the right amount of messages', async () => {
-      const response = await lambda().invoke({
+      const { Payload } = await lambda().invoke({
         FunctionName: `${config.stackName}-sqs2sfThrottle`,
         InvocationType: 'RequestResponse',
         Payload: JSON.stringify({
           queueUrl: maxQueueUrl,
-          messageLimit: numberOfMessages,
-          timeLimit: 1
+          messageLimit: totalNumMessages
         })
       }).promise();
-
-      console.log(response);
-      const { Payload } = response;
-
       messagesConsumed = parseInt(Payload, 10);
-      console.log('messages consumed', messagesConsumed);
       // Can't test that the messages consumed is exactly the number the
       // maximum allowed because of eventual consistency in SQS
       expect(messagesConsumed).toBeGreaterThan(0);
```

Comment:
So the amount of messages it would consume (up to the message limit passed to the Lambda) is unpredictable here as well. Before, I did have this written as:

`expect(messagesConsumed).toBeLessThanOrEqual(queueMaxExecutions)`

However, there are some cases where the Lambda consumer starting executions runs longer than the lifespan of these executions (5 seconds), so the semaphore is decremented for executions that finish and the consumer is able to start more executions. This happens pretty infrequently, but it does happen. 

Given these kind of possible timing inconsistencies, I decided that the only important time to enforce the maximum execution limit is respected is when we count the number of running executions. What do you think?

Reply:
Enforcing that the number of running executions respects an execution limit sounds like the right path to take.
Taking a small step back, what is the intent of the `expect(messagesConsumed).toBeGreaterThan(0)` line?

Reply:
Yeah, it's clearly not testing very much other than that at least **some amount** messages were consumed by the lambda from the queue. But given the inability to predict how many or guarantee that the number will be less than the executions max, the test can't really be much more precise

Reply:
Is there any value in changing that `0` to a higher value? Correct me if I'm wrong, but we should be able to say with certainty that _at least_ the maximum executions number of messages should be read off of the queue and started, right?

Reply:
Unfortunately, we cannot say that with certainty, because SQS is eventually consistent and reading from it right after you've put messages in the queue may not give you that exact number of messages. This is also why the test above only checks that the number of messages consumed is greater than 0

Reply:
Sounds good to me. Thanks for walking through that with me. The final suggestion I have, then, would be to change the string of this `it` block from `consumes the right amount of message` to something like `consumes some number of messages`. This isn't a blocking change, but it makes the test more readable to me.